### PR TITLE
Closes #2855: Full-width background padding

### LINKED ISSUE #2855: Left alignment issue between content blocks
## Problem/Motivation

There is an alignment inconsistency occurring between content blocks when building pages. Is it possible to make it so that all text is consistently left aligned for all content blocks? I am not sure why this is happening, perhaps there is an easy solution that I am missing.

image

## Proposed resolution
Description of the proposed solution, the rationale behind it, and workarounds for people who cannot use the patch.

##  Screenshots
If applicable, add screenshots to help explain your problem. (Need a tool to share screenshots or screencasts as URLs? Try [monosnap](https://monosnap.com)

### PR DESCRIPTION
**Pull request by danahertzberg**
## Description
Looking to align content within a full width background to other content on the page. Add demo content to cover these testing needs.

## Related issues
Closes #2855 

## How to test

1. Add full-width background to pages with and without sidebar nav
   a. `/pages/combo-page`
   b. `/pages/combo-page-no-sidebar`
4. Verify main content and full-width content align at widths

https://094aab32-6c3a-47f7-8953-f5415d7033d0.probo.build/

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [x] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by HenryzGreenberg** [comment]
@mrbeiley Can we see what we can do about when you add full with background the side nav gets pushed way down.  https://app.screencast.com/G9e5YLp32ZCle

https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2861.probo.build/pages/text-background

**Comment by danahertzberg** [comment]
> @mrbeiley Can we see what we can do about when you add full with background the side nav gets pushed way down. https://app.screencast.com/G9e5YLp32ZCle
> 
> https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2861.probo.build/pages/text-background

This is expected behavior. New solutions welcomed.

File: modules/custom/az_paragraphs/az_paragraphs_text_background/templates/paragraph--az-text-background--default.html.twig [hunk 1]
```diff
@@ -57,8 +57,8 @@
     {% block content %}
       {# Layout for content width AND full width styles #}
       <div class="container">
-        <div class="az-full-width-row">
-          <div class="az-full-width-column-content">
+        <div class="">
+          <div class="">
```
Comment (joshuasosa) [inline modules/custom/az_paragraphs/az_paragraphs_text_background/templates/paragraph--az-text-background--default.html.twig:61]: Are these div's still needed? They could probably just be deleted.

**Comment by mrbeiley** [comment]
Basic, local question: if I change a .twig file, then run 'lando start', will the changes be reflected on my local site?

**Comment by danahertzberg** [comment]
> Basic, local question: if I change a .twig file, then run 'lando start', will the changes be reflected on my local site?
> 
> or is there maybe a better way to test changes?

@mrbeiley If you want to make local changes to the twig file, you can clear cache after the site is built and that should be enough. Otherwise re-running lando install (not lando start) will rebuild the site from scratch.

**Comment by mrbeiley** [comment]
Seems like the Full-width content and Main content are lined up with and without Nav bars

image

image

**Comment by joshuasosa** [comment]
@mrbeiley I've noticed that too, but probably not directly related to this PR or #2855. I think those notes would be good to add as two new issues in the queue. #1827 might be related.

**Review by mrbeiley — COMMENTED** [review]
Looks good to me.

**Review by mrbeiley — DISMISSED** [review]

**Review by joshuasosa — DISMISSED** [review]
Looks ok on my testing using Probo and a local install for comparison. I couldn't really tell a difference, which is good.

**Comment by mrbeiley** [comment]
Viewing not in menu at all: looks good on my laptop screen.

image

**Comment by danahertzberg** [comment]
Interesting.. thanks for the review! Looking at it in today's meeting will be helpful. 🙏

**Comment by danahertzberg** [comment]
https://github.com/az-digital/az_quickstart/assets/59745109/9b86e1a1-e1dd-41d3-b99f-a4bad1c44be4

**Review by sanjanans — APPROVED** [review]

**Comment by mrbeiley** [comment]
> Need to review with media aspect ratio. Review "janky-ness" with video. Also review on multidev (@camikazegreen).
> 
> May need to consider CSS updates rather than removing the classes.

What needs to be done, if anything?

**Comment by danahertzberg** [comment]
- Mac settings > Appearance > Always show scrollbars = possible cause?
- Content area is shifted to the right when using full-width text with background paragraph type (not non-full-width).
- Needs more testing.

**Comment by danahertzberg** [comment]
To do
- [ ] Add demo content to display full-width and text areas together (assigned to Dana)

**Comment by danahertzberg** [comment]
Demo content is added! And I'm seeing this problem. Does anyone else see it?
https://accb7534-856e-4f47-8fd9-0c261fb40dce.probo.build/pages/combo-page-no-sidebar

image

**Comment by joshuasosa** [comment]
> Demo content is added! And I'm seeing this problem with alignment on pages without sidebar. Does anyone else see it? https://accb7534-856e-4f47-8fd9-0c261fb40dce.probo.build/pages/combo-page-no-sidebar
> 
Looks like `.region-content .full-width-background` is making text go to the left a bit more than it should. I'm not currently sure how to fix.

**Comment by danahertzberg** [comment]
Thank you for finding a cause @joshuasosa! I'll look into it in the morning.

**Comment by danahertzberg** [comment]
It's actually this causing the misalignment. Getting rid of it allows for horizontal scroll. Not sure what to do about it yet.

```
.full-width-background {
    max-width: calc(100vw - var(--scrollbar-width) / 2);
}
```

**Comment by joshuasosa** [comment]
There may have been updates in the past three years where it's no longer needed.

Another option might be to replace `.region-content .full-width-background` with:
```
.region-content .full-width-background {
    margin-left: var(--full-width-left-distance);
    margin-right: var(--full-width-right-distance);
    overflow: hidden;
}
```
(Replicates change [here](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_paragraphs/css/az_paragraphs_full_width.css#L143).)

And change `.full-width-background` to:
`max-width: calc(100vw - var(--scrollbar-width));`
(Replicates change [here](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_paragraphs/css/az_paragraphs_full_width.css#L20).)

**Comment by danahertzberg** [comment]
THANK YOU @joshuasosa!! That fixed it locally for me. I'll make sure probo is good now too. But this should then be complete finally!

**Comment by danahertzberg** [comment]
PROBO WORKS!! https://98ba7673-228b-4a78-a870-2125441660d5.probo.build/pages/combo-page-no-sidebar

**Comment by bberndt-uaz** [comment]
Unfortunately, the change to the `.full-width-background` CSS rule causes the center line of alternating full-width/non-full-width split screen paragraphs to be misaligned when no sidebar is present: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2861.probo.build/brian-test-page

**Comment by joshuasosa** [comment]
> Unfortunately, the CSS changes to `.full-width-background` cause the center line of alternating full-width/non-full-width split screen paragraphs to be misaligned when no sidebar is present: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2861.probo.build/brian-test-page

Nice catch! It looks like the standard-width split screen adds a right margin.
Removing `main:only-child .az-splitscreen:not(.full-width-background)` from [az_splitscreen.css](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_paragraphs/az_paragraphs_splitscreen/css/az_splitscreen.css#L21) seems to fix it.

**Comment by danahertzberg** [comment]
Thanks again @joshuasosa! Looks like paragraphs are working well now.

- https://094aab32-6c3a-47f7-8953-f5415d7033d0.probo.build/pages/dana
- https://094aab32-6c3a-47f7-8953-f5415d7033d0.probo.build/pages/clone-dana

**Review by joeparsons — APPROVED** [review]

**Review by trackleft — APPROVED** [review]
Nice work!

**Review by joshuasosa — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Review by sanjanans — APPROVED** [review]

**Review by mrbeiley — APPROVED** [review]

**Pull request merged**